File: regex_extracting/extracting/regex_lib/PhoneAttachment.py
# encoding=utf8
"""
    Author:  'jdwang'
    Date:    'create date: 2017-01-13'; 'last updated date: 2017-01-13'
    Email:   '383287471@qq.com'
    Describe:
"""
from __future__ import print_function
from regex_extracting.extracting.common.regex_base import RegexBase

__version__ = '1.0'


class PhoneAttachment(RegexBase):
    name = '手机附件'

    def __init__(self,sentence):

        # 要处理的输入句子

        self.sentence = sentence

        # region 1 初始化正则表达式
        # 描述正则表达式
        self.statement_regexs = [
            '手机附件|附件'
        ]
        # 值正则表达式
        self.value_regexs = [
            '主机|锂电池|数据线|耳机|充电器|说明书',
               '随意|随便|都可以|其他|别的'
        ]
        # endregion
        super(PhoneAttachment, self).__int__()
        self.regex_process()


if __name__ == '__main__':
    price = PhoneAttachment('价不高')
    for info_meta_data in price.info_meta_data_list:
        print('-'*80)
        print(str(info_meta_data))

File: regex_extracting/asp_version/Extracting/PhoneAttachment.cs
﻿/*
 * author : wxp
 * date : 2013/9/11
 * */

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using System.Web;

namespace Ch2R.Code.Extracting
{
    public class PhoneAttachment : ParameterBase
    {
        public PhoneAttachment(Areas.Chat.Models.UserSession userSession, string sentence)
            : base(userSession, sentence)
        {
            name = "手机附件";

            #region 初始化正则表达式
            // 描述正则表达式
            StatementRegexs = new string[]
            {
                @"手机附件|附件"    
            };
            // 值正则表达式
            ValueRegexs = new string[]
            {
                @"主机|锂电池|数据线|耳机|充电器|说明书",
               @"随意|随便|都可以|其他|别的"
            };
            #endregion

            RegexProcess();
        }

        public override void Format()
        {
            for (int i = 0; i < list.Count; ++i)
            {
                if (Regex.IsMatch(list[i].rawData, @"随意|随便|都可以"))
                {
                    list[i].normalData = "";
                }
                else if (Regex.IsMatch(list[i].rawData, @"别的|其他的"))
                {
                    if (userSession.liveTable.Parameter["手机附件"] != null)
                    {
                        list[i].normalData = (String)userSession.liveTable.Parameter["手机附件"];
                        list[i].isNegative = true;
                    }
                    else
                    {
                        /**
                         * 仍有其他价格为null的情况
                         * 别的未测试
                         * */
                    }
                }
                else
                {
                    list[i].normalData = list[i].rawData;
                }
                /* 正常人都不会嫌附件多
                 * 所以这里不做否定形式处理
                 */
            }
        }

        public override InfoBlock ToInfoBlock()
        {
            List<InfoUnit> L = new List<InfoUnit>();
            foreach (InfoMetadata im in list)
            {
                InfoUnit iu = new InfoUnit();
                iu.normalData = im.normalData;
                iu.rawData = im.rawData;
                iu.isNegative = im.isNegative;
                L.Add(iu);
            }
            InfoBlock ib = new InfoBlock(name, "and", L);
            /* 这里关系应该是“and”
             * 用户对附件的要求一般是同时要求多个的
             */
            return ib;
        }
    }
}